cpg_utils/config-template.toml

```diff
@@ -65,59 +76,23 @@ check_intermediates = true
 # already exist. If it exists, do not submit stage jobs
 check_expected_outputs = true
 
-# Local directory for temporary files. Usually takes a few kB.
-# If not provided, a temp folder will be created
-#local_tmp_dir =
-
-# Slack channel to send status and QC reports:
-#[slack]
-#channel = 'seqr-loader-test'
-#token_secret_id = 'slack-seqr-loader-token'
-#token_project_id = 'seqr-308602'
-
-# Prefix to find docker images referenced in `image_config_yaml_path`
-image_registry_prefix = 'australia-southeast1-docker.pkg.dev/cpg-common/images'
-
-# Prefix to find reference files referenced in `refdata_yaml_path`
-reference_prefix = 'gs://cpg-reference'
-
-# Template to build HTTP URLs matching the dataset_path of category
-# "web". Should be parametrised by namespace and dataset in Jinja format:
-web_url_template = 'https://{namespace}-web.populationgenomics.org.au/{dataset}'
-
-# GCP project to activate and for requester-pays buckets. For CPG, will be determined
-# automatically if not provided, and analysis-runner service account is activated.
-#dataset_gcp_project =
-
 # Limit to data of this sequencing type
 #sequencing_type = 'genome'
 
-# Realign CRAM when available, instead of using FASTQ.
-# The parameter value should correspond to CRAM version
-# (e.g. v0 in gs://cpg-fewgenomes-main/cram/v0/CPG01234.cram
-#realign_from_cram_version = 'v0'
-
-# Use GnarlyGenotyper instead of GenotypeGVCFs
-use_gnarly = false
-
-# Use allele-specific annotations for VQSR
-use_as_vqsr = true
-
-# Calling intervals (defauls to whole genome intervals)
-#intervals_path =
-
-# Create Seqr ElasticSearch indices for these datasets. If not specified, will
-# create indices for all input datasets.
-#create_es_index_for_datasets = []
-
-[elasticsearch]
-# Configure access to ElasticSearch server
-port = 9243
-host = 'elasticsearch.es.australia-southeast1.gcp.elastic-cloud.com'
-username = 'seqr'
-# Load ElasticSearch password from a secret, unless SEQR_ES_PASSWORD is set
-password_secret_id = 'seqr-es-password'
-password_project_id = 'seqr-308602'
+# Templates to construct output storage paths. Used by `cpg_utils.hail_batch` function
+# `output_path`. The scheme below implements CPG storage policy
+# https://github.com/populationgenomics/team-docs/tree/main/storage_policies
+path_scheme = 'cpg_google_storage'
+
+[path_schemes.cpg_google_storage]
```

Comment:
Do we really need schemes here? Wouldn't it be easier to understand if these were just direct prefixes, but configured by the analysis-runner / some manual config?

E.g. why not simply `tmp = "gs://cpg-tob-wgs-main-tmp"` when you run a pipeline on the TOB-WGS dataset with a prod access level?

Reply:
@lgruen, I think I'm following, but just to be sure, can you show an example of a config like you see it, and example of `output_path` call?

Reply:
@lgruen do you mean that at runtime the config should just contain exact paths, e.g. a run using analysis-runner & the output flag `-o pinocchio` would enter `paths` into the config dictionary

```toml
[paths]
default="gs://cpg-tob-wgs-main/pinocchio"
tmp="gs://cpg-tob-wgs-main-tmp/pinocchio"
web="gs://cpg-tob-wgs-main-web/pinocchio"
...
```

these paths would still need to be configured in analysis-runner according to a set schema, even if they weren't made on the fly with each `output_path` call

Reply:
Adding suffixes at runtime sounds fine (and required for the way we use `dataset_path` at the moment), but I was just wondering whether we really need the scheme for the bucket itself. I.e. the following part should just be fixed for any given invocation, right?
```toml
[paths]
default="gs://cpg-tob-wgs-main"
tmp="gs://cpg-tob-wgs-main-tmp"
web="gs://cpg-tob-wgs-main-web"
...
```

Reply:
But that would be a lot of repetition. Plus, for some workflows we don't have just one dataset, e.g. for seqr-loader we would write to `gs://cpg-acute-care-main`, `gs://cpg-perth-neuro-main`, etc. Should that be redesigned as well?

Reply:
Great point about workflows that write to multiple dataset locations in a single run. I hadn't thought of that.

I think this means we should also replace the current `dataset_path` implementation with your more flexible approach.

Reply:
> I think this means we should also replace the current dataset_path implementation with your more flexible approach.
Sorry, I'm slow today. What do you mean? This PR updates `dataset_path` to use the suggested approach.